concourse/client/util.py

```diff
@@ -122,3 +152,159 @@ def jobs_not_triggered(
             for pr_resource_version in pr_resource_versions:
                 if not wait_for_job_to_be_triggered(job, pr_resource_version, concourse_api):
                     yield (job, pr_resource, pr_resource_version)
+
+
+def get_resource_lock_and_trigger_build(
+    job: Job,
+    resource: PipelineConfigResource,
+    resource_version: ResourceVersion,
+    concourse_api,
+    max_retries: int=3,
+):
+    retries = max_retries
+    while retries >= 0:
+        try:
+            with pin_resource(
+                resource=resource,
+                resource_version=resource_version,
+                concourse_api=concourse_api,
+            ):
+                concourse_api.trigger_build(
+                    pipeline_name=resource.pipeline_name(),
+                    job_name=job.pipeline_name,
+                )
+
+                job_retry = 10
+                job_sleep_seconds = 5
+                if wait_for_job_to_be_triggered(
+                    job=job,
+                    resource_version=resource_version,
+                    concourse_api=concourse_api,
+                    retries=job_retry,
+                    sleep_time_seconds=job_sleep_seconds,
+                ):
+                    info(
+                        f'job {job.name} for resource version {resource_version.version()} '
+                        'has been triggered'
+                    )
+                else:
+                    pin_comment = PinComment(
+                        pin_timestamp=datetime.now().isoformat(),
+                        version_data=resource_version.version(),
+                        next_retry=(datetime.now() + timedelta(
+                            seconds=(job_retry + 1) * job_sleep_seconds,
+                        )).isoformat(),
+                        comment="pinned by technical user, please do not unpin manually"
```

Comment:
can we please add our username?

Reply:
also, the comment should be changed to: do not unpin before (<earliest date after which it is safe to unpin>)